core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -1343,6 +1343,13 @@ public static ShardId selectSplitShard(int shardId, IndexMetaData sourceIndexMet
                  + "] must be less that the number of target shards [" + numTargetShards + "]");
         }
         int routingFactor = getRoutingFactor(numSourceShards, numTargetShards);
+        // now we verify that the numRoutingShards is valid in the source index
+        int routingNumShards = sourceIndexMetadata.getRoutingNumShards();
+        int factor = routingNumShards / numTargetShards;
+        if (factor * numTargetShards != routingNumShards) {
```

Comment:
the division then multiplication again looks like you actually want to check that `routingNumShards % numTargetShards != 0`?